### PR TITLE
Make PAYGAS an invalid opcode in most cases

### PR DESCRIPTION
**Pull request by SamWilsn**

### CODE REVIEW

File: core/vm/jump_table.go [hunk 1]
```diff
@@ -47,6 +47,7 @@ type operation struct {
 	valid   bool // indication whether the retrieved operation is valid and known
 	reverts bool // determines whether the operation reverts state (implicitly halts)
 	returns bool // determines whether the operations sets the return data content
+	intern  bool // indicates that the operation is safe to call before paygas
```
Comment (lightclient) [inline core/vm/jump_table.go]: Is `intern` short for anything?

Comment (SamWilsn) [reply core/vm/jump_table.go]: I was going for "internal reads/writes only".

Comment (lightclient) [reply core/vm/jump_table.go]: I see, that seems reasonable. I think `internal` is short enough it might be worth spelling out for clarity.

**Review by adietrichs — APPROVED** [review]
slightly misleading title: PAYGAS outside AA is a no-op, not an invalid opcode

**Review by lightclient — APPROVED** [review]
ty :)

**Pull request merged**